src/infrastructure/Page.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * It represents a page of results after a repository search call.
+ *
+ * @param <T> then model type.
+ */
+export class Page<T> {
```

Comment:
I would transform this page to a more TS POJO like the other objects making the fields public read-only. This style is "too Java" and inconsistent with the TS current mode and could have issues if converting it from/to json.

isLast could be an attribute calculated and set in the constructor so it can be serialized/deserialized too.